go/vt/vtgate/engine/batch_route.go

```diff
@@ -0,0 +1,126 @@
+package engine
```

Comment:
This sounded like a good idea when we discussed. But now I see that it makes everything awkward. I think the code will be much simpler if this was done in router.go. Just iterate, get plan & execute.

This approach will probably make sense if we decide to do the phases, but I think we may never do that if the parallel thing works out.